### DISCUSSION TITLE
Setting value and control heating circuits via json using a "Select" or even a "Climate" entity?

### DISCUSSION THREAD

**Discussion by aschix** [dominikamann/oekofen-pellematic-compact#50] [Ideas]
For now the integration is fine for reading values usind a `Sensor` and a `Binary Sensor`.
It would however be fantastic to be able to use the integration to write/set values.
First of all I think about the relevant fields of an heating circuit, e.g. `hk1.mode_auto` (using a `Select` in ha?) via `hk1.temp_heat` (`Number`?)
I guess it should even be possible to make out an entire `Climate` entity for every circuit

Has anyone thoughts about these things as well?

**Comment by dominikamann** [comment]
Hi @aschix 

See https://github.com/dominikamann/oekofen-pellematic-compact/discussions/28

Unfortunately, I haven't had time to implement this yet. You are welcome to help develop it 😉

**Reply by aschix** [reply]
Thanks! Ok, #28 is about the states (select), I'd have a further look if it would be in principle possible to combine every to a climate, as mentioned. Unfortunately I do not come with experience in ha-python programming, but i can have a look - but i am afraid not in the near future :(

**Comment by dominikamann** [comment]
@aschix It is now possible, thanks to @sweetcorreze !
See release: https://github.com/dominikamann/oekofen-pellematic-compact/releases/tag/v3.3.0

**Comment by dominikamann** [comment]
See also https://github.com/dominikamann/oekofen-pellematic-compact/issues/80

**Comment by pdruart** [comment]
If someone is willing to help, I’ve developed something that works quite well, but I could use some assistance to optimize and finalize it.
https://gist.github.com/pdruart/f6e844331a0e773e980dc146264f5013

**Comment by dominikamann** [comment]
See https://github.com/dominikamann/oekofen-pellematic-compact/releases/tag/v3.5.1

**Comment by Horsti-Knut** [comment]
Hello there,
i'm using your integration with my pellematic smart xs and it is very helpfull.
I'm missing only the Main mode from the system...
My json code is:
`...{
 "system":{
  "L_ambient":"186", 
  "L_errors":"0", 
  "L_usb_stick":"false", 
  "L_existing_boiler":"0", 
  "mode":"2"
 },
 "weather":{ ...`

The main mode is 0=off, 1=winter (all circuits on) and 2 is summer (only hot water)
If thats a select option in home assistant, that would be great and i'm happy.

Thanks!

**Reply by dominikamann** [reply]
Hi @Horsti-Knut,

do you mean steering or reading?

Thanks

**Reply by Horsti-Knut** [reply]
Hi @dominikamann,
i mean steering. Is it possible?
--> "mode":"2"

Hallo @dominikamann,
Steuerung wäre toll, da es offensichtlich über json übermittelt wird und wohl auch steuerbar ist. Es ist ja kein "L_" vorm Datensatz des json.
--> "mode":"2"

**Reply by dominikamann** [reply]
See https://github.com/dominikamann/oekofen-pellematic-compact/issues/134
